src/dongyeon-lee/230127/P67256.kt

```diff
@@ -0,0 +1,34 @@
+package `dongyeon-lee`.`230127`
+
+import kotlin.math.abs
+
+const val STAR = 10 ; const val HASH = 11 // *, #
+const val RIGHT_HAND = "right"
+class Solution {
+    fun solution(numbers: IntArray, hand: String): String {
+        val phone = listOf(
+            (4 to 2),
+            (1 to 1), (1 to 2), (1 to 3),
+            (2 to 1), (2 to 2), (2 to 3),
+            (3 to 1), (3 to 2), (3 to 3),
+            (4 to 1),           (4 to 3)
+        )
+        val hands = mutableListOf(phone[STAR], phone[HASH])
+        return numbers.map { num ->
+            val numLoc = phone[num]
+            val leftDst = abs(hands[0].first - numLoc.first) + abs(hands[0].second - numLoc.second)
+            val rightDst = abs(hands[1].first - numLoc.first) + abs(hands[1].second - numLoc.second)
+            when {
+                num in listOf(3,6,9) -> "R"
+                num in listOf(1,4,7) -> "L"
+                leftDst < rightDst -> "L"
+                leftDst > rightDst -> "R"
+                hand == RIGHT_HAND -> "R"
+                else -> "L"
```

Comment:
```suggestion
                else -> "R".takeIf { hand == RIGHT_HAND } ?: "L"
```

너무 takeIf 를 좋아하남 ㅎㅎ,,

Reply:
원래는
enum class HAND(val dir){
    LEFT_HAND("L"),
    RIGHT_HAND("R")
}

선언해놓구,
val handType = HAND.valueOf(hand) 만든담에

else -> handType.dir
써도 좋겠다 생각은 했었는데,, 괜스레 복잡해지는 느낌이예요.